[doc, CI] doc driven CI and docker flow refactor, bugfix (#1312)
radixark/miles: docs/ci/01-label.md

## Meta-labels: run everything

`run-ci-image` and `run-ci-all` are the same switch: both add `--match-all-labels` to run every enabled test in the suite, ignoring domain labels. Neither is in `KNOWN_LABELS`.

A manual `workflow_dispatch` run gets `--match-all-labels` too, because it has no PR labels to filter on.

## Registration and scan scope

Labels are optional; registration is not. The runner scans `tests/fast`, `tests/fast-gpu`, `tests/e2e`, `tests/ci` recursively for `test_*.py`. Every file must resolve to a registration or collection fails:

- A file outside `tests/fast/` with no `register_*_ci()` call → `No CI registry found`.
- A `labels=[...]` value not in `KNOWN_LABELS` → `unknown labels [...]`.

## `tests/fast/` auto-registers as CPU

Each `test_*.py` under `tests/fast/` is auto-registered as a CPU test (backend CPU, suite `stage-a-cpu`, `labels=[]`) with no `register_*_ci()` call, and runs on the GitHub-hosted `ubuntu-latest` runner. Here "CPU" is the hardware backend, not a label. A `register_cuda_ci()` under `tests/fast/` is a hard error — move it to `tests/fast-gpu/`.

## `bypass-fastfail`: opt out of fast-fail

By default CI fails fast on two levels:

> - Cross-stage: GPU stages skip if `stage-a-cpu` fails (`!failure()` in their `if`).

Reviewer: 01-label.md documents a !failure() gate that doesn't exist.
  ▎ L58/L63 say the cross-stage gate is !failure(), but grep '!failure()' pr-test.yml finds
  ▎ nothing. The real gate (pr-test.yml:113-118) uses needs.stage-a-cpu.result == 'success'.
  ▎ These differ: !failure() is true when stage-a-cpu is skipped; == 'success' is not. Workflow
  ▎ is fine — just fix the doc.

  Fix — docs/ci/01-label.md

  L58:
  ```suggestion
  - Cross-stage: GPU stages run only when `stage-a-cpu` succeeds — the `if` requires
  `needs.stage-a-cpu.result == 'success'`.
  ```

  L63:
  ```suggestion
  - Cross-stage: each GPU stage's check becomes `(needs.stage-a-cpu.result == 'success' ||
  (needs.stage-a-cpu.result == 'failure' && contains(..., 'bypass-fastfail')))`, so GPU stages
  run even after `stage-a-cpu` fails.
  ```

  from cc for your reference
